tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsWithOptionsFacts.cs

```diff
@@ -0,0 +1,323 @@
+
+            Assert.DoesNotContain("Generate type 'NonExistentBaseType' -> Generate new type...", response);
+        }
+
+        [Fact]
+        // Theres generate type without options that gives ~same result with default ui. No point to bring this.
+        public async Task Blacklists_pull_members_up_with_UI()
+        {
+            const string code =
+                @"
+                public class Class1: BaseClass
+                {
+                    public string Foo[||] { get; set; }
+                }
+
+                public class BaseClass {}
+";
+
+            var response = await FindRefactoringNamesAsync(code);
+
+            Assert.DoesNotContain("Pull 'Foo' up -> Pull members up to base type...", response);
+        }
+
+        [Fact]
+        public async Task Can_extract_interface()
+        {
+            const string code =
+                @"public class Class1[||]
+                {
+                    public string PropertyHere { get; set; }
+                }";
+
+            const string expected =
+                @"
+                public interface IClass1
+                {
+                    string PropertyHere { get; set; }
+                }
+
+                public class Class1 : IClass1
+                {
+                    public string PropertyHere { get; set; }
+                }
+                ";
+            var response = await RunRefactoringAsync(code, "Extract Interface...");
+
+            var foo = ((ModifiedFileResponse)response.Changes.First()).Buffer;
+            AssertIgnoringIndent(expected, ((ModifiedFileResponse)response.Changes.First()).Buffer);
+        }
+
+        [Fact]
+        public void Check_better_alternative_available_for_codeaction_with_options()
+        {
+            // This keeps record is Microsoft.CodeAnalysis.PickMembers.IPickMembersService public or not
+            // and should fail on case where interface is exposed. Which likely means that this is officially
+            // supported scenario by roslyn.
+            //
+            // In case it's exposed by roslyn team these services can be simplified.
+            // Steps are likely following:
+            // - Remove proxies
+            // - Replace ExportWorkspaceServiceFactoryWithAssemblyQualifiedName with Microsoft.CodeAnalysis.Host.Mef.ExportWorkspaceServiceAttribute
+            // - Fix proxy classes to implement IPickMembersService / IExtractInterfaceOptionsService ... instead of proxy and reflection.
+            // - Remove all factories using ExportWorkspaceServiceFactoryWithAssemblyQualifiedName and factory itself.
+            // Following issue may have additional information: https://github.com/dotnet/roslyn/issues/33277
+            var pickMemberServiceType = Assembly.Load("Microsoft.CodeAnalysis.Features").GetType("Microsoft.CodeAnalysis.PickMembers.IPickMembersService");
+            Assert.False(pickMemberServiceType.IsPublic);
```

Comment:
this test is a very good idea 👏

Reply:
Well, unless they change the API shape before making it public :)

Reply:
we could add the check for the parameters too, although we don't do it for the current reflected APIs - I think we could handled that in one sweep separately